src/overlays/actors/ovl_En_Fu/z_en_fu.c

```diff
@@ -27,33 +39,274 @@ const ActorInit En_Fu_InitVars = {
     (ActorFunc)EnFu_Update,
     (ActorFunc)EnFu_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fu/EnFu_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fu/EnFu_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x11, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 30, 40, 0, { 0, 0, 0 } },
+};
+
+static Vec3f sMtxSrc = {
+    700.0f,
+    700.0f,
+    0.0f,
+};
+
+static UNK_PTR sEyesSegments[] = {
+    0x06005F20,
+    0x06006320,
+};
+
+static UNK_PTR sMouthSegments[] = {
+    0x06006720,
+    0x06006920,
+};
+
+extern AnimationHeader D_0600057C;
+extern AnimationHeader D_06000B04;
+extern SkeletonHeader D_06006C90;
+extern UNK_TYPE D_0200E080;
+
+typedef enum {
+    /* 0x00 */ FU_FACE_CALM,
+    /* 0x01 */ FU_FACE_MAD,
+} EnFuFace;
+
+void EnFu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFu* this = THIS;
+    s32 pad;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 36.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelanime, &D_06006C90, &D_06000B04, this->limbDrawTable,
+                     this->transitionDrawTable, 16);
+    SkelAnime_ChangeAnimDefaultRepeat(&this->skelanime, &D_06000B04);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 0xFF;
+    Actor_SetScale(&this->actor, 0.01f);
+    if (LINK_IS_CHILD) {
+        this->actionFunc = EnFu_WaitChild;
+        this->facialExpression = FU_FACE_CALM;
+    } else {
+        this->actionFunc = EnFu_WaitAdult;
+        this->facialExpression = FU_FACE_MAD;
+        this->skelanime.animPlaybackSpeed = 2.0f;
+    }
+    this->behaviorFlags = 0;
+    this->actor.unk_1F = 6;
+}
+
+void EnFu_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFu* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+s32 func_80A1D94C(EnFu* this, GlobalContext* globalCtx, u16 textID, EnFuActionFunc actionFunc) {
+    s16 yawDiff;
+
+    // func_8002F194 returns 1 if actor flags & 0x100 is set and unsets it
+    if (func_8002F194(&this->actor, globalCtx)) {
+        this->actionFunc = actionFunc;
+        return 1;
+    }
+    this->actor.textId = textID;
+    yawDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;
+
+    if ((ABS(yawDiff) < 0x2301) && (this->actor.xzDistanceFromLink < 100.0f)) {
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    } else {
+        this->behaviorFlags |= FU_RESET_LOOK_ANGLE;
+    }
+    return 0;
```

Comment:
I believe true and false can be used here
```suggestion
        return true;
    }
    this->actor.textId = textID;
    yawDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;

    if ((ABS(yawDiff) < 0x2301) && (this->actor.xzDistanceFromLink < 100.0f)) {
        func_8002F2CC(&this->actor, globalCtx, 100.0f);
    } else {
        this->behaviorFlags |= FU_RESET_LOOK_ANGLE;
    }
    return false;
```